packages/parameters/src/appconfig/AppConfigProvider.ts

```diff
@@ -321,7 +355,7 @@ class AppConfigProvider extends BaseProvider {
   protected async _getMultiple(
     _path: string,
     _sdkOptions?: unknown
-  ): Promise<Record<string, string | undefined>> {
+  ): Promise<void> {
```

Comment:
Also here, unclear why it is `void`

Reply:
Similar to [here](https://github.com/aws-powertools/powertools-lambda-typescript/pull/1528#discussion_r1238421232), `void` is used when there's no return. This method always throws an error.

Looking at these two functions:
```ts
const functionA = (): void => {
  console.log('hi');
}

const functionB = (): undefined => {
  console.log('hi');

  return;
}
```

In the example above, `functionA` never returns anything, as far as TypeScript is concerned. The other one, `functionB` instead returns an `undefined` (empty = `undefined`).

In JS terms, the return is implicit, so in reality both return `undefined`, however this tells the TS compiler that the function is not expected to return anything at all.

Using `undefined`, tells the caller that there is a value, and this value is `undefined`, so I can do `const result = functionB();` and know that `result` here is undefined.

Conversely, if I did `const restulA = functionA();`, I would get an error because `functionA()` is not expected to return any value.